### PR TITLE
[On Hold] Default icon generation tool.

### PR DESCRIPTION
**Pull request by pogo-excalibur**
## Description
This adds a Python script for generating default Pokemon icons.

## Motivation and Context
Gen 3 is likely to start arriving sometime around Halloween, and at present RM has no (automattic) way of generating the necessary default icons.

This tool allows us to quickly generate more default icons, both immediately (for Gen3) and in the future (for later generations and forms).

## How Has This Been Tested?
Icon generation was run on a Ubuntu 16.04 server.

## Types of changes

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: Tools/Icon-Generation/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+This is a tool for generating RocketMap's base Pok�mon icons.
+
+## Install
+
+To use this tool you need:
+* Pillow - a python image library (a replacement for PIL)
+* The Arial Bold truetype font
+
+To install pillow run `pip install pillow`.
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]: Just add `pillow` to `requirements.txt`.

File: Tools/Icon-Generation/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+## Description
+
+This is a tool for generating RocketMap's base Pok�mon icons.
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]: > to generate

File: Tools/Icon-Generation/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Install
+
+To use this tool you need:
+* Pillow - a python image library (a replacement for PIL)
+* The Arial Bold truetype font
+
+To install pillow run `pip install pillow`.
+
+Installation of the Arial Bold font varies by platform. In Windows, it will typically be available by default. For linux systems, you may need to install the package `ttf-mscorefonts-installer` (or the equivallent package for your platform).
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]: Typo:

> equivalent

Also add the related command to install `ttf-mscorefonts-installer` and  mention macOS.

File: Tools/Icon-Generation/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+## Usage
+
+To generate the standard 94x94 icons and save them to the `static/icons/` directory, run:
+
+`generate_icons.py`
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]:
```
python generate_icons.py
```

File: Tools/Icon-Generation/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+To set a different output location, set the `-o` parameter:
+
+`generate_icons.py -o OUTPUT_DIR`
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]: Same as above, prefix with `python`.

File: Tools/Icon-Generation/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+To generate icons of a different size, set the `-s` parameter:
+
+`generate_icons.py -s SIZE`
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/README.md]: Same as above, prefix with `python`.

File: Tools/Icon-Generation/generate_icons.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
+import os
+
+from argparse import ArgumentParser
+from json import load as load_json
+
+import PIL
+from PIL import Image
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/generate_icons.py]: You `import PIL` and also `from PIL import Image`. Later on, you reference `PIL.Image`. Just pick one of the two import types.

File: Tools/Icon-Generation/generate_icons.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    'arial_bold_44': ImageFont.truetype('arialbd.ttf', 44)
+}
+
+
+def create_icon(id, form=None, size=(94, 94), base_size=(94, 94),
+                background_color='#000000', text_color='#FFFFFF'):
+    id = str(id) if id else id
+    form = str(form) if form else form
+
+    icon = PIL.Image.new('RGBA', (base_size[0], base_size[1]))
```
Comment (sebastienvercammen) [inline Tools/Icon-Generation/generate_icons.py]:
```
from PIL import Image

icon = Image.new(...)
```

**Comment by pogo-excalibur** [comment]
!! This PR is currently on hold as this functionality is likely to get merged into #2348. !!

**Comment by pogo-excalibur** [comment]
This has now been merged into #2348.

**Pull request closed without merging**